evm_arithmetization/src/prover.rs

```diff
@@ -560,6 +576,23 @@ pub fn check_abort_signal(abort_signal: Option<Arc<AtomicBool>>) -> Result<()> {
     Ok(())
 }
 
+/// Returns a vector containing the data required to generate all segments for
+/// one full execution.
+pub fn generate_all_data_segments<F: RichField>(
+    max_cpu_len_log: Option<usize>,
+    inputs: GenerationInputs,
+) -> anyhow::Result<Vec<GenerationSegmentData>> {
+    // TODO: change this value after some benchmarks have been carried out?
+    let mut all_data = Vec::with_capacity(50);
+    let mut running = true;
+    while running {
+        let res = generate_segment_given_data::<F>(max_cpu_len_log, inputs.clone(), &mut all_data)?;
+        if res.is_none() {
+            running = false;
+        }
+    }
```

Comment:
I'd tend to prefer something concise like 

```suggestion
    while generate_segment_given_data::<F>(max_cpu_len_log, &inputs, &mut all_data)?.is_some() {
        // Iterate until we have generated all segments.
    }
```

but up to you